src/tests/Microsoft.PowerFx.Interpreter.Tests/ConfigTests.cs

```diff
@@ -82,6 +82,33 @@ public void DisplayNameBinding()
             Assert.True(check.IsSuccess);
         }
 
+        [Theory]
+        [InlineData("displayVariable + 1", "logicalVariable + 1", 2)]
+        [InlineData("logicalVariable + 1", "logicalVariable + 1", 2)]
+        [InlineData("If(true, logicalVariable)", "If(true, logicalVariable)", 1)]
+        public async void TopLevelVariableDisplayName(string expression, string expectedInvariantExpression, double expected)
+        {
+            var symbol = new SymbolTable();
+
+            // Adds display name for a variable.
+            var logicalVariableSlot = symbol.AddVariable("logicalVariable", FormulaType.Number, displayName: "displayVariable");
+            var r1 = new SymbolValues(symbol);
+            r1.Set(logicalVariableSlot, FormulaValue.New(1));
+
+            var config = new PowerFxConfig() { SymbolTable = symbol };
+
+            var engine = new RecalcEngine(config);
+
+            var check = engine.Check(expression);
+            Assert.True(check.IsSuccess);
+
+            var eval = await engine.EvalAsync(expression,CancellationToken.None, runtimeConfig: r1);
+            Assert.Equal(expected, eval.ToObject());
+
+            var actualInvariantExpression = engine.GetInvariantExpression(expression, null);
```

Comment:
>engine.GetInvariantExpression [](http://example.com/codeflow?start=44&length=29)

Can we run other direction too and test engine.GetDisplay?

Reply:
updated.